src/libraries/System.IO.Hashing/src/System/IO/Hashing/Crc32.X86.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel.DataAnnotations;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics.X86;
+using System.Runtime.Intrinsics;
+using System.Runtime.InteropServices;
+
+namespace System.IO.Hashing
+{
+    public partial class Crc32
+    {
+        private const int X86BlockSize = 64;
```

Comment:
After considering it, I agree. The constant was a holdover from a previous iteration where I was checking the length before calling the method, which made it harder to intuit the value. Since you requested that the length check be moved to the `Update` method, I've left the constant for that purpose only and renamed it appropriately. All the other sites use `Vector128<byte>.Count`.

Let me know if you still think the `Update` method should just use `Vector128<byte>.Count * 4`. I just thought it made things clearer when the logic is split between two files.